src/utils/dayViewLayout.js

```diff
@@ -254,6 +257,26 @@ export default function getStyledEvents ({
       });
     });
 
+    // If the previous parent event had children, we need to check if any of its children
+    // overlap with the current parent event. If so, we need to set the number of columns
+    // to the previous number of columns to make space for the overlapping child.
+    if (prevParentHasChildren) {
+      const hasOverlapWithPrevParentChild = events.some((_event, eventIdx) => {
+        if (eventIdx >= idx) return false;
+        return isChild(eventIdx, idx, { events, startAccessor, endAccessor, min, totalMin });
+      });
+      if(hasOverlapWithPrevParentChild) {
+        nbrOfColumns = prevNbrOfColumns;
+      }
+    }
+    prevNbrOfColumns = nbrOfColumns;
+
+    if (childGroups.length) {
+      prevParentHasChildren = true;
+    } else {
+      prevParentHasChildren = false;
+    }
```

Comment:
Can we move these `prev` value assignments to the end of the `while` loop? Putting them in the middle here is error-prone because any code past this point is going to be using the wrong "prev" value.